ingestors/nuget.go

```diff
@@ -0,0 +1,132 @@
+package ingestors
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/librariesio/depper/data"
+)
+
+const nugetSchedule = "*/5 * * * *"
+const nugetIndexUrl = "https://api.nuget.org/v3/catalog0/index.json"
+
+type nugetIndex struct {
+	IndexId string `json:"@id"`
+	Pages   []struct {
+		Url             string `json:"@id"`
+		CommitTimeStamp string `json:"commitTimeStamp"`
+		CommitTime      time.Time
+	} `json:"items"`
+}
+
+type nugetPage struct {
+	PageId   string `json:"@id"`
+	Packages []struct {
+		Url             string `json:"@id"`
+		CommitTimeStamp string `json:"commitTimeStamp"`
+		CommitTime      time.Time
+		Name            string `json:"nuget:id"`
+		Version         string `json:"nuget:version"`
+	} `json:"items"`
+}
+
+type Nuget struct {
+	LatestRun time.Time
+}
+
+func NewNuget() *Nuget {
+	return &Nuget{}
+}
+
+func (ingestor *Nuget) Schedule() string {
+	return nugetSchedule
+}
+
+func (ingestor *Nuget) Ingest() []data.PackageVersion {
+	// Until we save LatestRun state, go back to a 4x multiple of the frequency: 20 min.
+	if ingestor.LatestRun.IsZero() {
+		ingestor.LatestRun = time.Now().Add(-20 * time.Minute)
```

Comment:
Ignore all my other doofy comments. My one comment is, maybe just make this a static 2 hour duration, a little extra buffer won't kill us.

Reply:
Constant rather